### ISSUE TITLE
'NoneType' object has no attribute 'set' error

### ISSUE THREAD

**Issue by wirthandras** [WeblateOrg/weblate#6672]
Hi, We are using XLIFF v1.2 format via git source control.
The intermediate language comes from branch "X" and the output branch is "Y" 

The source review process run out without reported problem. (Intermediate language to English US)
But at Spanish translation process the user get an error. (English US -> Spanish US)

No clue what user did exactly. Some of texts has XML markup.
We assume the User saved one or more string with incorrect xml format.

If we try to save this text on Spanish(US) version the error occured:
```xliff
simple text: <x id="START_LINK" ctype="x-a" equiv-text="&lt;a href=&quot;https://validurl.com/&quot;&gt;"/>https://validurl.com/<x id="CLOSE_LINK" ctype="x-a" equiv-text="&lt;/a&gt;"/> 
```
This text saved without problem at source review.

An other string has an "<" which is first character of an incorrect xml part in Spanish translation

Some texts which has markup could be not saved, the errors come up.
The simple texts which has no any special formatting can be saved.

The biggest pain, it seems any of xml exports does not work, we cannot do a commit and push back to repository. (Same error occurs)
We tested the manual JSON export, it is works as expected.

**I already tried**

We tried to delete the problematic texts from weblate manually to expecting the xml exporters come back to work, but without success.
When we do a delete on the specific text the same error occurs

Environment: Docker compose install: Weblate version 4.6.1
**Exception traceback**

Request Method: POST
Request URL: https://url/commit/project/component/

Django Version: 3.2
Python Version: 3.7.3
Installed Applications:
['customize',
 'weblate.addons',
 'weblate.auth',
 'weblate.checks',
 'weblate.formats',
 'weblate.glossary',
 'weblate.machinery',
 'weblate.trans',
 'weblate.lang',
 'weblate_language_data',
 'weblate.memory',
 'weblate.screenshots',
 'weblate.fonts',
 'weblate.accounts',
 'weblate.configuration',
 'weblate.utils',
 'weblate.vcs',
 'weblate.wladmin',
 'weblate.metrics',
 'weblate',
 'weblate.gitexport',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'django.contrib.admin.apps.SimpleAdminConfig',
 'django.contrib.admindocs',
 'django.contrib.sitemaps',
 'django.contrib.humanize',
 'social_django',
 'crispy_forms',
 'compressor',
 'rest_framework',
 'rest_framework.authtoken',
 'django_filters']
Installed Middleware:
['weblate.middleware.RedirectMiddleware',
 'weblate.middleware.ProxyMiddleware',
 'django.middleware.security.SecurityMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'weblate.accounts.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'social_django.middleware.SocialAuthExceptionMiddleware',
 'weblate.accounts.middleware.RequireLoginMiddleware',
 'weblate.api.middleware.ThrottlingMiddleware',
 'weblate.middleware.SecurityMiddleware']

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.7/dist-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.7/dist-packages/django/contrib/auth/decorators.py", line 21, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/django/views/decorators/http.py", line 40, in inner
    return func(request, *args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/weblate/trans/views/git.py", line 138, in commit_component
    return perform_commit(request, obj)
  File "/usr/local/lib/python3.7/dist-packages/weblate/trans/views/git.py", line 60, in perform_commit
    request.user,
  File "/usr/local/lib/python3.7/dist-packages/weblate/trans/views/git.py", line 35, in execute_locked
    result = call(*args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/weblate/trans/models/component.py", line 154, in on_link_wrapper
    return func(self, *args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/weblate/trans/models/component.py", line 1543, in commit_pending
    reason, user, skip_push=True, force=True, signals=False
  File "/usr/lib/python3.7/contextlib.py", line 74, in inner
    return func(*args, **kwds)
  File "/usr/local/lib/python3.7/dist-packages/weblate/trans/models/translation.py", line 540, in commit_pending
    self.update_units(units, store, author_name, author.id)
  File "/usr/local/lib/python3.7/dist-packages/weblate/trans/models/translation.py", line 666, in update_units
    pounit.set_state(unit.state)
  File "/usr/local/lib/python3.7/dist-packages/weblate/formats/ttkit.py", line 680, in set_state
    self.xliff_node.set("state", "needs-translation")

Exception Type: AttributeError at /commit/princess-house/consultant-corner-frontend/
Exception Value: 'NoneType' object has no attribute 'set'

**Server configuration and status**

Weblate installation: Docker 

Any idea? Could we get out from stucked state?

Thanks!

**Comment by nijel** [comment]
> Any idea?

Please upgrade, this should be already fixed by b6d4c5794bd4f4fcfc856a6121f5675ceee9ee10.

**Comment by nijel** [comment]
Can you please share traceback from 4.8.1?

**Comment by wirthandras** [comment]
@nijel additional info: we dindt do any magic, but the automatic push to branch is works, but I dont know why, only the manual commit broken, we did a fresh install on a sandbox with 4.8.1, and it seems the problem not occurs. Could be something stucked?

**Comment by pilcrowls** [comment]
Hello there, I had the exact same problem using weblate 4.14.1-1 (last version as of today).
I tried to investigate a bit but did not find anything special. 
It fails when importing a valid (checked) XLF file. 
Reseting the underlying repository works but that way it is then not possible to import the given file. 
The solution I used is to reinstall everything and now with the same condition it works. 
So my guess would be some sort of data corruption or similar.
